ada_feeding/ada_feeding/watchdog/estop_condition.py

```diff
@@ -0,0 +1,421 @@
+            ParameterDescriptor(
+                name="device",
+                type=ParameterType.PARAMETER_INTEGER,
+                description=(
+                    "The index of the audio device to use. This should usually "
+                    "be 0, to indicate the default audio device."
+                ),
+                read_only=True,
+            ),
+        )
+        self.device = device.value
+
+        initial_wait_secs = self._node.declare_parameter(
+            "initial_wait_secs",
+            2.0,
+            ParameterDescriptor(
+                name="initial_wait_secs",
+                type=ParameterType.PARAMETER_DOUBLE,
+                description=(
+                    "The number of seconds to wait before checking if the e-stop "
+                    "button is plugged in. This is because there is typically initial "
+                    "noise from the audio signal."
+                ),
+                read_only=True,
+            ),
+        )
+        self.initial_wait_duration = Duration(seconds=initial_wait_secs.value)
+
+        acpi_event_name = self._node.declare_parameter(
+            "acpi_event_name",
+            "jack/microphone MICROPHONE",
+            ParameterDescriptor(
+                name="acpi_event_name",
+                type=ParameterType.PARAMETER_STRING,
+                description=(
+                    "The name of the ACPI event that is triggered when the e-stop "
+                    "button is (un)plugged. Get this value by running `acpi_listener` "
+                    "in the terminal and unplugging the e-stop button. Exclude the word "
+                    "` plug` or ` unplug` from the end of the event name."
+                ),
+                read_only=True,
+            ),
+        )
+        self.acpi_event_name = acpi_event_name.value
+
+        min_threshold = self._node.declare_parameter(
+            "min_threshold",
+            -10000,
+            ParameterDescriptor(
+                name="min_threshold",
+                type=ParameterType.PARAMETER_INTEGER,
+                description=(
+                    "A falling edge must go below this threshold to be considered "
+                    "a button click"
+                ),
+                read_only=True,
+            ),
+        )
+        self.min_threshold = min_threshold.value
+
+        max_threshold = self._node.declare_parameter(
+            "max_threshold",
+            10000,
+            ParameterDescriptor(
+                name="max_threshold",
+                type=ParameterType.PARAMETER_INTEGER,
+                description=(
+                    "A rising edge must go above this threshold to be considered "
+                    "a button click"
+                ),
+                read_only=True,
+            ),
+        )
+        self.max_threshold = max_threshold.value
+
+        time_per_click_sec = self._node.declare_parameter(
+            "time_per_click_sec",
+            0.5,
```

Comment:
Not something that needs to be changed now, but let's keep an eye out during testing to make sure that 0.5 sec is enough time for most/all users.